engine/runtime-parser/src/main/java/org/enso/compiler/core/ir/Type.java

```diff
@@ -0,0 +1,158 @@
+package org.enso.compiler.core.ir;
+
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.module.scope.Definition;
+import org.enso.runtime.parser.dsl.GenerateFields;
+import org.enso.runtime.parser.dsl.GenerateIR;
+import org.enso.runtime.parser.dsl.IRChild;
+import org.enso.runtime.parser.dsl.IRField;
+import scala.Option;
+import scala.collection.immutable.List;
+
+public interface Type extends Expression {
+  @Override
+  Type mapExpressions(java.util.function.Function<Expression, Expression> fn);
+
+  @Override
+  Type setLocation(Option<IdentifiedLocation> location);
+
+  @Override
+  Type duplicate(
+      boolean keepLocations,
+      boolean keepMetadata,
+      boolean keepDiagnostics,
+      boolean keepIdentifiers);
+
+  @GenerateIR(interfaces = {Type.class})
+  final class Function extends TypeFunctionGen {
+    @GenerateFields
+    public Function(
+        @IRChild List<Expression> args,
```

Comment:
- looks like this `Type.Function` IR element is using [n-ary representation](https://github.com/enso-org/design/pull/60)
- CCing @kazcw